app/controllers/declaration/FiscalInformationController.scala

```diff
@@ -60,29 +60,32 @@ class FiscalInformationController @Inject()(
       .bindFromRequest()
       .fold(
         (formWithErrors: Form[FiscalInformation]) => Future.successful(BadRequest(fiscalInformationPage(mode, itemId, formWithErrors))),
-        formData =>
-          formData.onwardSupplyRelief match {
-            case FiscalInformation.AllowedFiscalInformationAnswers.yes =>
-              updateCacheForYes(itemId, formData) map { _ =>
-                navigator.continueTo(routes.AdditionalFiscalReferencesController.displayPage(mode, itemId))
-              }
-            case FiscalInformation.AllowedFiscalInformationAnswers.no =>
-              updateCacheForNo(itemId, formData) map { _ =>
-                navigator.continueTo(routes.CommodityDetailsController.displayPage(mode, itemId))
-              }
-        }
+        formData => updateCache(itemId, formData).map(_ => redirectToNextPage(mode, itemId, formData))
       )
   }
 
-  //TODO Use one method instead of updateCacheForYes and updateCacheForNo
-  private def updateCacheForYes(itemId: String, updatedFiscalInformation: FiscalInformation)(implicit req: JourneyRequest[AnyContent]): Future[Unit] =
-    updateExportsDeclarationSyncDirect(model => {
-      model.updatedItem(itemId, item => item.copy(fiscalInformation = Some(updatedFiscalInformation)))
-    }).map(_ => ())
+  private def updateCache(itemId: String, updatedFiscalInformation: FiscalInformation)(
+    implicit request: JourneyRequest[AnyContent]
+  ): Future[Option[ExportsDeclaration]] = {
+    val updateModel: ExportsDeclaration => ExportsDeclaration =
+      (model: ExportsDeclaration) =>
+        updatedFiscalInformation.onwardSupplyRelief match {
+          case FiscalInformation.AllowedFiscalInformationAnswers.yes =>
+            model.updatedItem(itemId, item => item.copy(fiscalInformation = Some(updatedFiscalInformation)))
```

Comment:
Nit: Maybe the `item` could be extracted into a variable just before passing into the update?

Reply:
Item is different for Yes and for No, so I'm not sure that building this outside make sense

Reply:
What do you think about move item copy part as dedicated method on `ExportItem` ?

Reply:
Logic here is slightly connected with the user choice, so we're updating differently for Yes and for No. Not sure that there is value to moving this to the item, but I can do it